DEFI/LiquidityPool/certora/specs/Full.spec

```diff
@@ -0,0 +1,448 @@
+/***
+This example is a full spec for LiquidityPool.
+To run this use Certora cli with the conf file runFullPoll.conf
+Example of a run: https://prover.certora.com/output/1512/8fb4697781554144ac1e137fed6ca241?anonymousKey=e8a8cf904d540a3f0f8306574c3cee72dadf8d6c
+UnsatCores: https://prover.certora.com/output/1512/d76148722d6d4da98d4b050d032f3ae1/UnsatCoreVisualisation.html?anonymousKey=0df6a5352e20529c1b97a1a6d2846d9a25207b35
+Mutation test for this spec: https://mutation-testing.certora.com/?id=1dd7e71d-1451-48a9-ae17-d05a46b1581e&anonymousKey=79e13549-03d2-4436-8293-0c7d0b6b20d9
+See https://docs.certora.com for a complete guide.
+***/
+
+using Asset as underlying;
+using TrivialReceiver as _TrivialReceiver;
+
+methods
+{
+    function balanceOf(address)                      external returns(uint256) envfree;
+    function totalSupply()                           external returns(uint256) envfree;
+    function transfer(address, uint256)              external returns(bool);
+    function transferFrom(address, address, uint256) external returns(bool);
+    function amountToShares(uint256)                 external returns(uint256) envfree;
+    function sharesToAmount(uint256)                 external returns(uint256) envfree;
+    function deposit(uint256)                        external returns(uint256);
+    function withdraw(uint256)                       external returns(uint256);
+    function calcPremium(uint256)                    external returns (uint256) envfree;
+
+    function _.executeOperation(uint256,uint256,address) external => DISPATCHER(true);
+    function flashLoan(address, uint256)                 external;
+
+    function underlying.balanceOf(address)               external returns(uint256) envfree;
+    function underlying.allowance(address, address)      external returns(uint256) envfree;
+    function underlying.totalSupply()                    external returns(uint256) envfree;
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Reentrancy ghost and hook                                                                                           │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+ghost uint256 ghostReentrancyStatus;
+
+hook Sload uint256 status currentContract._status STORAGE {
+    require ghostReentrancyStatus == status;
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Deposite                                                                                                            │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule depositIntegrity(env e){
+
+    require e.msg.sender != currentContract; // this assumption must hold to avoid shares dilute attack
```

Comment:
need to say what it is a safe assumption, because we prove the contract does not call itself in rule ... , thus the msg.sender can not be the current LiquidityPool

Reply:
Didnt prove yet, still looking for the rule

Reply:
lets add later